pandas/core/dtypes/cast.py

```diff
@@ -2208,10 +2208,11 @@ def can_hold_element(arr: ArrayLike, element: Any) -> bool:
 
     elif dtype.kind == "f":
         if tipo is not None:
-            # TODO: itemsize check?
             if tipo.kind not in ["f", "i", "u"]:
                 # Anything other than float/integer we cannot hold
                 return False
+            elif dtype.itemsize < tipo.itemsize:
```

Comment:
should the itemsize only be compared if dtype.kind and tipo.kind are the same? i.e. tipo.kind is also 'f'. How is the itemsize of an int or uint relevant?

Reply:
Thanks for bringing this up. I think it ultimately comes down to -> if `can_hold_element` holds True, then using `np.putmask` is assumed to be safe. So itemsize of int or uint matters because `numpy` allows safe casting of i16 to f32 for example, but not i64 to f32.

But I think the itemsize check is not sufficient here, because `numpy` won't `putmask` `int32` in`float32`.  Maybe `np.can_cast` should be used instead?